game-core/src/main/java/games/strategy/engine/lobby/client/login/LobbyServerPropertiesFetcher.java

```diff
@@ -101,21 +106,18 @@
     lobbyProps.ifPresent(props -> {
       ClientSetting.lobbyLastUsedHost.setValue(props.getHost());
       ClientSetting.lobbyLastUsedPort.setValue(props.getPort());
-      ClientSetting.lobbyLastUsedHttpHostUri.setValue(props.getHttpServerUri());
+      ClientSetting.lobbyLastUsedHttpsPort.setValue(props.getHttpsPort());
       ClientSetting.flush();
     });
     return lobbyProps;
   }
 
   private static Optional<LobbyServerProperties> getLastUsedProperties() {
-    if (ClientSetting.lobbyLastUsedHost.isSet()
-        && ClientSetting.lobbyLastUsedPort.isSet()
-        && ClientSetting.lobbyLastUsedHttpHostUri.isSet()) {
-
+    if (ClientSetting.lobbyLastUsedHost.isSet()) {
       return Optional.of(LobbyServerProperties.builder()
           .host(ClientSetting.lobbyLastUsedHost.getValueOrThrow())
-          .port(ClientSetting.lobbyLastUsedPort.getValueOrThrow())
-          .httpServerUri(ClientSetting.lobbyLastUsedHttpHostUri.getValueOrThrow())
+          .port(ClientSetting.lobbyLastUsedPort.getValue().orElse(TEST_LOBBY_DEFAULT_PORT))
+          .httpsPort(ClientSetting.lobbyLastUsedHttpsPort.getValue().orElse(TEST_LOBBY_DEFAULT_HTTPS_PORT))
           .build());
```

Comment:
As already pointed out by you, previously you had to explicitly change all properties for them to be picked up, now the settings only get picked up if the host has changed (even if the ports were changed).
Wouldn't it be better if all settings would work independently? Not sure if this is in the scope of this PR.

Reply:
The settings never worked independently as they all had to be set. That was actually a point of confusion @ssoloff pointed out, the http lobby URI had to also be set along with the lobby host override for any of the override values to work.

> now the settings only get picked up if the host has changed (even if the ports were changed).

The ports are ignored unless the host value is set. The ports are not required, if the host value is set then default values will be used for ports.

Reply:
Yeah, I know that. Previously it was just that _all_ of the settings had to be set, all of them were equally important, and now only the host is important

Reply:
how do you mean equally important? If only one was set, it was ignored. That was actually a point of error, setting 3 properties seemed counter-intuitive when just updating the local lobby host.

Reply:
@RoiEXLab this update makes it very easy to test lobby, you just have to set the host override to 'localhost' and you're done. Before it was a PITA, had to set that, the port, *and* the spark server override.

Are you advocating that we go back to having to specify multiple data points for the override to work? I'm not sure I understand the concern here.

Reply:
I didn't write my points out well.
I'm not saying that the "previous"/current system was great, I'm just saying that I'd prefer a consistency so that I can change the port without having to change the host as well.

Reply:
I see the point of having the values work independently:
- i'm not sure we want to support that use-case
- it's more complexity
- this update gets us closer to that and not necessarily further away

For live lobbies, the live properties file is meant to be the mechanism to switch between the lobbies. I'm quite not happy we send 1.10 clients to the 1.9 lobby and it's dangerous frankly as 1.10 games can crash 1.9 games.

The use-case for the test override is mostly to allow for local testing. I'm almost tempted to convert these override values to a simple checkbox of 'use-local'.